### ISSUE TITLE
cs:~natefinch/starsay is not returning latest published revision

### ISSUE THREAD

**Issue by natefinch** [juju/charmstore#617]
```
$ charm push . cs:~natefinch/starsay 
url: cs:~natefinch/starsay-3
channel: unpublished
$ charm publish ~natefinch/starsay-3 --resource data-0
url: cs:~natefinch/starsay-3
channel: stable
warning: bugs-url and homepage are not set.  See set command.
$ juju deploy cs:~natefinch/starsay
Added charm "cs:~natefinch/starsay-0" to the model.
Deploying charm "cs:~natefinch/starsay-0" with the default charm metadata series "trusty".
```

see also: https://api.jujucharms.com/charmstore/v5/meta/any?channel=stable&id=cs:~natefinch/starsay&include=id-revision

**Comment by mhilton** [comment]
This looks to be a problem with the resolution of multi-series charms where different versions of the charms support different series.

**Comment by mhilton** [comment]
This is an interesting one, if I repeatedly load the above URL it sometimes returns the correct result. It looks like the order in which we iterate through the map is important.

**Comment by mhilton** [comment]
Fixed in #629
